src/double64.c

```diff
@@ -519,7 +519,7 @@ d2i_clip_array (const double *src, int count, int *dest, double scale)
 
 		if (CPU_CLIPS_POSITIVE == 0 && tmp > (1.0 * INT_MAX))
 			dest [count] = INT_MAX ;
-		else if (CPU_CLIPS_NEGATIVE == 0 && tmp < (-1.0 * INT_MAX))
+		else if (CPU_CLIPS_NEGATIVE == 0 && tmp < (1.0 * INT_MIN))
```

Comment:
The existing code was as expected. The reason is that the existing code was applying clipping symmetrically (same clip value for positive and negative values).

Reply:
If this was your purpose, then I think the statement
`dest [count] = INT_MIN ;`
should became
`dest [count] = -INT_MAX ;`
, should it ?

Reply:
That may well be the case. I added an issue for it https://github.com/erikd/libsndfile/issues/324